app/models/vendor_code.rb

```diff
@@ -7,11 +7,8 @@ class VendorCode < ActiveRecord::Base
                    length: { maximum: 255, message: "longer than 255 characters" }
 	
 	
-  def assign_to(current_user)
-    self.update_attributes(user: current_user, user_name: current_user.name, email: current_user.email)
-    vendor = self.vendor
-    vendor.update_attribute(:usedCodes, vendor.usedCodes + 1)
-    vendor.update_attribute(:unclaimCodes, vendor.unclaimCodes - 1)
+  def associate_with(user)
+    self.update(user_id: user.id, user_name: user.name, email: user.email)
   end  
```

Comment:
Oleg, it is a sensible note.
From my point of view, it's not correct to change the number of codes until the vendor code is redeemed.
I think we can extract this question to the separate discussion.